Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.cpp

```diff
@@ -36,28 +41,22 @@ SettingVisibility NumericSettingBase::GetVisibility() const
   return m_details.visibility;
 }
 
-template <>
-void NumericSetting<int>::SetExpressionFromValue()
-{
-  m_value.m_input.SetExpression(ValueToString(GetValue()));
-}
-
-template <>
-void NumericSetting<double>::SetExpressionFromValue()
+template <typename T>
+void NumericSetting<T>::SetExpressionFromValue()
 {
-  // We must use a dot decimal separator for expression parser.
-  std::ostringstream ss;
-  ss.imbue(std::locale::classic());
-  ss << GetValue();
-
-  m_value.m_input.SetExpression(ss.str());
+  // Always include -/+ sign to prevent CoalesceExpression binding.
+  // e.g. 1 is a valid input name for keyboard devices, +1 is not.
+  // We must use "classic" locale. ExpressionParser expects dot decimal separator.
+  m_value.m_input.SetExpression(
+      fmt::format(std::locale::classic(), "{:+g}", ControlState(GetValue())));
```

Comment:
Except for the `L` option, `fmt::format` always uses the classic locale unless you specify otherwise (https://fmt.dev/latest/api.html#locale), so I think you can drop this.

Reply:
I'll change this.